### PR TITLE
Replace tasks list providers with id-based providing

### PR DESCRIPTION
**Pull request by bitfriend**
- Remove `ref.invalidate()` about parts that were added after [1st version](https://github.com/acterglobal/a3/pull/1598) of `subscribeStream` replacing
- Exclude dot file of flutter plugins from git

### CODE REVIEW

**Comment by bitfriend** [comment]
The following could fix issue that outer list page of tasklists is not updated when tasklist title changes:

1. Pass `taskListId` not `taskList` to `TaskListItemCard`, so that renderer can detect the clearer change condition
2. Convert `taskListId` into `taskList` using `ref.watch(taskListItemProvider(taskListId)).when()` in `TaskListItemCard`

Then `TaskListItemCard` in outer list page is automatically re-rendered whenever tasklist title changes.

**Review by kumarpalsinh25 — APPROVED** [review]
Code wise look good to me. At some of place not sure about  `ref.invalidate` removal so would request @gnunicorn to review once before it gets merged.

File: app/lib/features/tasks/widgets/task_item.dart [hunk 1]
```diff
@@ -11,47 +12,65 @@ import 'package:flutter_gen/gen_l10n/l10n.dart';
...
+            leading: leadingWidget(task),
+            title: takeItemTitle(context, task),
+            subtitle: takeItemSubTitle(ref, context, task),
+            trailing: trailing(ref, task),
+          ),
+          error: (error, stack) {
+            _log.severe('failed to load task', error, stack);
+            return const ListTile(
+              title: Text('failed'),
+            );
```
Comment (gnunicorn) [inline app/lib/features/tasks/widgets/task_item.dart:65]: we should show the error for easier debugging
```suggestion
            return ListTile(
              title: Text('Loading failed: $error'),
            );
```

File: app/lib/features/tasks/widgets/task_item.dart [hunk 2]
```diff
@@ -193,13 +212,11 @@ class TaskItem extends ConsumerWidget {
     );
   }
 
-  Key doneKey() {
-    final taskId = task.eventIdStr();
+  Key doneKey(String taskId) {
```
Comment (gnunicorn) [inline app/lib/features/tasks/widgets/task_item.dart:215]: these are not necessary. `taskId` is an attribute of the widget, we don't have to pass it in.

```suggestion
  Key doneKey() {
```

File: app/lib/features/tasks/widgets/task_item.dart [hunk 3]
```diff
@@ -193,13 +212,11 @@ class TaskItem extends ConsumerWidget {
...
     return Key('task-entry-$taskId-status-btn-done');
   }
 
-  Key notDoneKey() {
-    final taskId = task.eventIdStr();
+  Key notDoneKey(String taskId) {
```
Comment (gnunicorn) [inline app/lib/features/tasks/widgets/task_item.dart:219]:
```suggestion
  Key notDoneKey() {
```

File: app/lib/features/tasks/widgets/task_list_item_card.dart [hunk 4]
```diff
@@ -1,50 +1,64 @@
...
+            ),
           ),
-        ],
-      ),
-    );
+          error: (error, stack) {
+            _log.severe('failed to load tasklist', error, stack);
+            return const Card(
+              child: Text('failed'),
+            );
```
Comment (gnunicorn) [inline app/lib/features/tasks/widgets/task_list_item_card.dart:53]: same as above.

File: app/lib/features/tasks/widgets/task_list_item_card.dart [hunk 5]
```diff
@@ -1,50 +1,64 @@
...
+          },
+          loading: () => const Card(
+            child: Text('loading'),
+          ),
+        );
   }
 
-  Widget title(BuildContext context) {
+  Widget title(BuildContext context, TaskList taskList) {
     final taskListId = taskList.eventIdStr();
```
Comment (gnunicorn) [inline app/lib/features/tasks/widgets/task_list_item_card.dart:62]: unnecessary, we already know `taskListId`.
```suggestion
  Widget title(BuildContext context, TaskList taskList) {
```

**Review by gnunicorn — APPROVED** [review]
It's fine but the title of the PR doesn't describe this ...

File: app/lib/features/tasks/widgets/task_item.dart (hunk 1 above)
Comment (bitfriend) [reply app/lib/features/tasks/widgets/task_item.dart]: Fixed by https://github.com/acterglobal/a3/pull/2046/commits/030800270c4512f1ac5892707630d8fdc0bc0b2d

File: app/lib/features/tasks/widgets/task_list_item_card.dart (hunk 4 above)
Comment (bitfriend) [reply app/lib/features/tasks/widgets/task_list_item_card.dart]: Fixed by https://github.com/acterglobal/a3/pull/2046/commits/030800270c4512f1ac5892707630d8fdc0bc0b2d

File: app/lib/features/tasks/widgets/task_item.dart (hunk 2 above)
Comment (bitfriend) [reply app/lib/features/tasks/widgets/task_item.dart]: Fixed by https://github.com/acterglobal/a3/pull/2046/commits/633f3a707a97476d9b2e667a0df11395dd3825cc

File: app/lib/features/tasks/widgets/task_item.dart (hunk 3 above)
Comment (bitfriend) [reply app/lib/features/tasks/widgets/task_item.dart]: Fixed by https://github.com/acterglobal/a3/pull/2046/commits/633f3a707a97476d9b2e667a0df11395dd3825cc

**Pull request merged**